test/TesterInternal/ReentrancyTests.cs

```diff
@@ -342,6 +482,55 @@ public void FanOut_AC_Stress_NonReentrant()
             Assert.True(elapsed < MaxStressExecutionTime, $"Stress test execution took too long: {elapsed}");
         }
     }
+
+    public class MessageInterleavesPredicateBootstrapProvider : IBootstrapProvider
+    {
+        public string Name { get; private set; }
+
+        public Task Init(string name, IProviderRuntime providerRuntime, IProviderConfiguration config)
+        {
+            Name = name;
+            InstallPredicate();
+            return TaskDone.Done;
+        }
+
+        void InstallPredicate()
+        {
+            GrainTypeData grainType;
+            GrainTypeManager.Instance.TryGetData(typeof(NonReentrantGrainWithMessageInterleavePredicate).FullName, out grainType);
+            grainType.SetMessageAllowedToInterleavePredicate(req =>
```

Comment:
I thought the plan was to provide an attribute or some other way to statically define the predicate.

Something like `[MessageAllowedToInterleavePredicate("UnitTests.Grains.NonReentrantGrainWithMessageInterleaveGrain.NonReentrantGrainWithMessageInterleavePredicate")]` on the grain implementation itself